src/main/java/com/lob/client/AsyncLobClient.java

```diff
@@ -532,12 +532,18 @@ public void onThrowable(final Throwable t) {
         public T onCompleted(final Response response) throws Exception {
             if (isSuccess(response)) {
                 final T value = MAPPER.readValue(response.getResponseBody(), clazz);
+
+                ((AbstractResponse) value).setStatusCode(response.getStatusCode());
+                ((AbstractResponse) value).setHeaders(response.getHeaders());
+
                 guavaFut.set(value);
                 return value;
             }
             else {
                 final ErrorResponse error = MAPPER.readValue(response.getResponseBody(), ErrorResponse.class);
                 final LobApiException exception = new LobApiException(response.getUri(), error);
+                exception.setStatusCode(response.getStatusCode());
+                exception.setHeaders(response.getHeaders());
```

Comment:
Thoughts on making them args to the constructor (and making them `final` in class) since we have them on initialization?